ios/brave-ios/Sources/AIChat/Components/Popover/PopoverView.swift

```diff
@@ -54,18 +54,43 @@ struct BravePopoverView<Content: View & PopoverContentComponent>: UIViewControll
 
   func updateUIViewController(_ uiViewController: UIViewController, context: Context) {
     if isPresented {
-      guard uiViewController.presentedViewController == nil, let parent = uiViewController.parent
+      guard uiViewController.presentedViewController == nil
       else {
+        // The system dismissed our popover automatically, but never updated our presentation state
+        // It usually does this if you present another popover or sheet
+        // Manually update it
+        if context.coordinator.presentedViewController != nil {
+          DispatchQueue.main.async {
+            isPresented = false
+          }
+        }
```

Comment:
Wouldn't this `updateUIViewController` only be called anyways if the binding/content inside changes? Does this code even hit ever unless you try to present again after its been auto-dismissed once?

Reply:
It gets called every time something happens. So if another controller gets presented on top of ours, this gets called for some odd reason, even if nothing changes.